include/oneapi/dpl/pstl/experimental/internal/for_loop_impl.h

```diff
@@ -472,64 +473,39 @@ __pattern_for_loop(_ExecutionPolicy&& __exec, _Ip __first, _Ip __last, _Function
 
 // Helper structure to split code functions for integral and iterator types so the return
 // value can be successfully deduced.
-template <typename _Ip, typename = void>
-struct __use_par_vec_helper;
-
 template <typename _Ip>
-struct __use_par_vec_helper<_Ip, ::std::enable_if_t<::std::is_integral_v<_Ip>>>
+struct __use_par_vec_helper
```

Comment:
It seems like this is used to accomplish the old-style `is_vector` and `is_parallel` as parameters for `__pattern_for_loop_n` (in the form of `std::true_type`, `std::false_type`).

I probably missed something unique to this pattern, but why does the tag dispatching pattern stop here instead of continuing into `__pattern_for_loop_n`?

Reply:
Previously we already had `clear` implementation of `for_each` staff with tags. But there are a lot of changes which difficult to review. So we made decision to have some implementation of `for_each` in the current PR with minimal amount of changes.